02week/ticTacToe.js

```diff
@@ -0,0 +1,203 @@
+      board[2][2] == playerTurn)
+  ) {
+    printBoard();
+    console.log("Player " + playerTurn + " wins!");
+    return true;
+  }
+}
+
+function diagonalWin() {
+  //checks to see if board array values satisfy a diagnonal win, print out board, announce winning player, and return true
+
+  if (
+    (board[0][0] == playerTurn &&
+      board[1][1] == playerTurn &&
+      board[2][2] == playerTurn) ||
+    (board[2][0] == playerTurn &&
+      board[1][1] == playerTurn &&
+      board[0][2] == playerTurn)
+  ) {
+    printBoard();
+    console.log("Player " + playerTurn + " wins!");
+    return true;
+  }
+}
+
+function checkForWin() {
+  //checkForWin merely checks to see if horizonalWin or verticalWin or diagonal win return true, and if so checkForWin returns true or false
+
+  if (horizontalWin() || verticalWin() || diagonalWin()) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function checkForDraw() {
+  /**checkForDraw checks to see if the board is filled up with no clear winner,
+   * its important that its at the bottom so that the file can check for a win first before checking for a draw.
+   * If a draw is detected it will print the board announce the draw and return true or false
+   **/
+  if (
+    board[0][0] !== " " &&
+    board[0][1] !== " " &&
+    board[0][2] !== " " &&
+    board[1][0] !== " " &&
+    board[1][1] !== " " &&
+    board[1][2] !== " " &&
+    board[2][0] !== " " &&
+    board[2][1] !== " " &&
+    board[2][2] !== " "
+  ) {
+    printBoard();
+    console.log("Its a draw!");
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function ticTacToe(row, column) {
+  //This is where most of the magic happens. There is some quality control going on here.
+  if (row > 2 || column > 2) {
+    //checks to see if row and column values from the user are under 2 and anounces a message.
+    console.log("Cordiantes must be between 0 and 2. Try again.");
+  } else if (row.length > 1 || column.length > 1) {
+    //checkes to see if row and column value length are over 1 character and announces a message.
+    console.log("Single digits please");
+  } //most of the work gets done here
+  else if (board[row][column] == " ") {
+    //checks to see if the board array cordinates are empty, everything within this else if statement will not run if that condition is not satisfied
+    board[row][column] = playerTurn; //assigns value if empty.
+    if (checkForWin() || checkForDraw()) {
+      //after player 'move' has been made this checks to see if there is a win or a draw and restarts the game as needed.
+      board[0][0] = " ";
+      board[0][1] = " ";
+      board[0][2] = " ";
+      board[1][0] = " ";
+      board[1][1] = " ";
+      board[1][2] = " ";
+      board[2][0] = " ";
+      board[2][1] = " ";
+      board[2][2] = " ";
+      playerTurn = "X";
+    } else if (
+      //if there is no winner or draw it immediately goes to check weather to switch to player X or O
```

Comment:
adding a comment in the middle of an if condition like this is really odd.
this would be better:
```else if(playerTurn == "X"){
      // if there is no winner or draw ....
      playerTurn = "O";
    } ````